### PR TITLE
fix(wallet/accounts): don't store derived account when it exists in app db

### LINKED ISSUE #10326: Can't add same account in wallet as the removed 'Watched only account'
# Bug Report

## Problem

Reported by user originally:
If you have `Watch only address` as multiaccount instance in Wallet - there is no way to add that account via seed phrase / private key even after `Watch only account` removed

#### Expected behavior
Can add account via seed phrase / private key  after the same account address removed which was watch only address

#### Actual behavior

Can NOT add account via seed phrase / private key  after the same account address removed which was watch only address

### Reproduction

1)  Open Status and create new account
2) In the `Wallet` -> `Add account` -> `Add watch only address` add the `0xB798BaA43Ea720dE862E96fefa9EDfd94273D6Db` address
3) Add this account by now as fully accessible account (`Add account` -> `Enter a seed phrase` -> use this seed phrase `lesson original rose sunset crane they document balcony lava jelly select december`
**Actual result = Expected result** `Account already exists` message appears and account is not added
4) Now remove that Watch only address added via it's detail view -> top-right `...` -> `Account settings` -> `Delete account` 
5) Add again this account by now as fully accessible account (`Add account` -> `Enter a seed phrase` -> use this seed phrase `lesson original rose sunset crane they document balcony lava jelly select december`

**Actual result** Account already exists` message appears and account is not added which is not expected since we deleted watch only acc before

**BIG HINT:** It's all fine if you do not add the same account while `Watch only address` (so if you repeat **Reproduction** but without 3rd step then all fine)

### Additional Information

- Status version: release1.2/develop nightly 13th of April
- Operating System: Android and iOS

#### Logs

```
04-14 16:40:44.951 17966 17966 D ViewRootImpl@fd7c74d[MainActivity]: ViewPostIme pointer 0
04-14 16:40:45.038 17966 17966 D ViewRootImpl@fd7c74d[MainActivity]: ViewPostIme pointer 1
04-14 16:40:45.063 17966 18008 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :wallet.accounts/add-new-account
04-14 16:40:45.067 17966 18008 D ReactNativeJS: DEBUG [status-im.wallet.accounts.core:230] - [wallet] add-new-account type :seed step nil
04-14 16:40:45.073 17966 18008 D ReactNativeJS: DEBUG [status-im.native-module.core:172] - [native-module] verify
04-14 16:40:45.074 17966 18009 D StatusModule: verify
04-14 16:40:45.934 17966 18008 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :wallet.accounts/add-new-account-password-verifyied
04-14 16:40:45.939 17966 18008 D ReactNativeJS: DEBUG [status-im.native-module.core:351] - [native-module] validate-mnemonic
04-14 16:40:45.939 17966 18009 D StatusModule: validateMnemonic
04-14 16:40:45.943 17966 20418 D StatusModule: {"error":""}
04-14 16:40:45.949 17966 18008 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :wallet.accounts/seed-validated
04-14 16:40:45.954 17966 18008 D ReactNativeJS: DEBUG [status-im.native-module.core:155] - [native-module] multiaccount-import-mnemonic
04-14 16:40:45.955 17966 18009 D StatusModule: multiAccountImportMnemonic
04-14 16:40:46.071 17966 18008 D ReactNativeJS: DEBUG [status-im.native-module.core:105] - [native-module]  multiaccount-derive-addresses
04-14 16:40:46.072 17966 18009 D StatusModule: multiAccountDeriveAddresses
04-14 16:40:46.082 17966 18008 D ReactNativeJS: DEBUG [status-im.native-module.core:131] - [native-module]  multiaccount-store-derived account-id 3ab1043a-8b75-476a-be78-9c9143028b2b
04-14 16:40:46.083 17966 18009 D StatusModule: multiAccountStoreDerived
04-14 16:40:46.091 17966 18073 E GoLog   : account already exists
04-14 16:40:46.099 17966 18008 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :status-im.wallet.accounts.core/new-account-error
04-14 16:40:46.386 17966 18073 D StatusModule: Signal event: {"type":"wallet","event":{"type":"newblock","blockNumber":9870907,"accounts":[],"newTransactions":{},"erc20":false}}
04-14 16:40:46.402 17966 18008 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :signals/signal-received
04-14 16:40:46.409 17966 18008 D ReactNativeJS: DEBUG [status-im.events:593] - :event-str {"type":"wallet","event":{"type":"newblock","blockNumber":9870907,"accounts":[],"newTransactions":{},"erc20":false}}
04-14 16:40:46.417 17966 18008 D ReactNativeJS: DEBUG [status-im.ethereum.subscriptions:96] - [wallet-subs] new-wallet-event event-type newblock
04-14 16:40:46.449 17966 18008 D ReactNativeJS: DEBUG [status-im.ethereum.subscriptions:33] - [wallet-subs] new-block accounts [] block 9870907 transactions-per-account {}
04-14 16:40:46.497 17966 17977 I zygote  : Background concurrent copying GC freed 166047(6MB) AllocSpace objects, 9(216KB) LOS objects, 49% free, 6MB/13MB, paused 441us total 108.245ms
04-14 16:40:46.756 17966 18008 D ReactNativeJS: DEBUG [status-im.utils.handlers:32] - Handling re-frame event:  :mailserver/check-connection-timeout
04-14 16:40:48.086 17966 17966 D ViewRootImpl@fd7c74d[MainActivity]: ViewPostIme pointer 0
04-14 16:40:48.464 17966 17966 D ViewRootImpl@fd7c74d[MainActivity]: ViewPostIme pointer 1



```

### PR DESCRIPTION
**Pull request by 0x-r4bbit**
### Summary

As discussed in #10326  and later in https://github.com/status-im/status-go/issues/1939, it turns out
that, when a user tries to add an account (from a seed phrase), while an account
with the same address (that the seed phrase would result in) exists in the application
state, the application would still go ahead and store the derived account in
status-go.

After that it still reports to the user that the 'Account already exits'.

This commit ensures that Status doesn't try to store the derived account when
the account already exists in the application database.

Fixes #10326

### Review notes

We need to check whether the address in question already exists in `multiaccount/accounts` so we need access to `db` inside `derive-and-store-account`. However since this is a normal function, that doesn't have access to it per se, I decided to pass down `accounts` from the outside before the function is called in the first place.

Wouldn't be surprised if there's a "cleaner" more re-frame-like way to do this. 

### Testing notes

Follow reproduction steps in #10326, should get expected results.

### CODE REVIEW

File: src/status_im/wallet/accounts/core.cljs [hunk 1]
```diff
@@ -157,8 +160,11 @@
   (let [error (:error (types/json->clj phrase-warnings))]
     (if-not (string/blank? error)
       (new-account-error cofx :account-error error)
-      {::import-account-seed {:passphrase      passphrase
-                              :hashed-password hashed-password}})))
+      (let [{:keys [db]} cofx
+            accounts (:multiaccount/accounts db)]
+        {::import-account-seed {:passphrase      passphrase
+                                :hashed-password hashed-password
+                                :accounts accounts }}))))
```
Comment (flexsurfer) [inline src/status_im/wallet/accounts/core.cljs]: a perfectionist is crying :)

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/wallet/accounts/core.cljs (hunk 1 above)
Comment (0x-r4bbit) [reply src/status_im/wallet/accounts/core.cljs]: As mentioned in the Review Notes, I'm very happy to iterate over this and change it to make it more reframe-y. Thoughts?

Comment (flexsurfer) [reply src/status_im/wallet/accounts/core.cljs]: i mean formatting, and btw lint has failed

**Comment by flexsurfer** [comment]
and also this `WARNING: Wrong number of args (3) passed to status-im.wallet.accounts.core/derive-and-store-account at line 118 /Users/jenkins/workspace/status-react_prs_ios_PR-10408/src/status_im/wallet/accounts/core.cljs`

**Comment by 0x-r4bbit** [comment]
Thanks for the pointer @flexsurfer totally forgot that `derive-and-store-account` was used multiple times. I've updated the PR with lint and formatting fixes.

**Comment by churik** [comment]
Tested:
- adding watch-only address and restoring it as main and vise versa
- adding account with seed phrase or private key and restoring it from seed phrase using main flow and vise versa (Multiaccount is already exists \ assount is already exist errors)

Looks good to me.

**Pull request merged**